monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -142,39 +148,56 @@ def get_scanners_of_machine(cls, machine: Machine) -> List[Machine]:
         for node in nodes:
             for dest, conn in node.connections.items():
                 if CommunicationType.SCANNED in conn and dest == machine.id:
-                    scanner_machine = ReportService._machine_repository.get_machine_by_id(
-                        node.machine_id
-                    )
+                    scanner_machine = cls._machine_repository.get_machine_by_id(node.machine_id)
                     scanner_machines.add(scanner_machine)
 
         return list(scanner_machines)
 
-    @staticmethod
-    def process_exploit(exploit) -> ExploiterReportInfo:
-        exploiter_type = exploit["data"]["exploiter"]
-        exploiter_descriptor = ExploiterDescriptorEnum.get_by_class_name(exploiter_type)
-        processor = exploiter_descriptor.processor()
-        exploiter_info = processor.get_exploit_info_by_dict(exploiter_type, exploit)
-        return exploiter_info
+    @classmethod
+    def process_exploit_event(
+        cls, exploit: ExploitationEvent, password_restored: Dict[ipaddress.IPv4Address, bool]
+    ) -> ExploiterReportInfo:
+        if not cls._machine_repository:
+            raise RuntimeError("Machine repository does not exist")
+
+        target_machine = cls._machine_repository.get_machines_by_ip(exploit.target)[0]
```

Comment:
I don't understand this logic. Why is the first machine always the one we want?

Reply:
IPs are not unique, but we're treating them like they are for now. This is an assumption we're making throughout the code until we can come up with a reliable way to disambiguate IP addresses.